app/models/archive_endpoint.rb

```diff
@@ -16,6 +16,23 @@ class ArchiveEndpoint < ApplicationRecord
   # TODO: after switching to string, validate that input resolves to class which #is_a class of the right type?
   validates :delivery_class, presence: true
 
+  # for the given druid, which archive endpoints should have archive copies?
+  scope :archive_targets, lambda { |druid|
+    joins(preservation_policies: [:preserved_objects]).where(preserved_objects: { druid: druid })
+  }
+
+  scope :which_have_archive_copy, lambda { |druid, version|
+    joins(archive_preserved_copies: [:preserved_object])
+      .where(
+        preserved_objects: { druid: druid },
+        archive_preserved_copies: { version: version }
+      )
+  }
+
+  scope :which_need_archive_copy, lambda { |druid, version|
```

Comment:
@SaravShah had suggested trying to work some bounds checking into the query, so that it doesn't return results for versions greater than the current version on the preserved object.  i'd like to try to do that, but it's not immediately clear to me what the query would be, so i have a slight preference for getting this merged and trying to do that as follow-on work.

if we did do that bounds checking in this query, it might or might not obviate the need for it in `PreservedCopy#create_archive_preserved_copies!`... i still sorta like the idea of failing noisily if an out of bounds version is given to that method, but i don't feel super-strongly about that.

Reply:
follow-on ticket sounds good to me.

Reply:
#983